### ISSUE TITLE
Performance of tokenizer for CLIP text model

### ISSUE THREAD

**Issue by michael-p** [huggingface/tokenizers#1412]
I'm the author of the [instant-clip-tokenizer](https://github.com/instant-labs/instant-clip-tokenizer) library which is a tokenizer specifically optimized for the CLIP text model.

I just ran a few benchmarks against the tokenizer from this library and it looks like our implementation is quite a bit faster (11x speed-up for batch encoding, 40x on single inputs), so I thought it might be interesting for you to look at the way we are doing the tokenization, maybe it helps you improve the speed of the tokenizer in this library.
Of course I understand that it is much more difficult to build and optimize a general purpose tokenizer (like you did) than one that works for "just" one model (like ours), but maybe you still find one or two things that help :)

The benchmark I used is below, am I missing something here or am I using the tokenizer in the wrong way? This is with version `4.35.2` of the `transformers` library.
<details>
  <summary>Benchmark code</summary>

```python
from transformers import CLIPTokenizerFast
from instant_clip_tokenizer import Tokenizer as InstantTokenizer

tokenizer_fast = CLIPTokenizerFast.from_pretrained("openai/clip-vit-base-patch16")
tokenizer_instant = InstantTokenizer()

INPUT = "a very typical bus station"

%timeit tokenizer_fast.encode(INPUT, add_special_tokens=False)
# -> 57.6 µs ± 174 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
%timeit tokenizer_instant.encode(INPUT)
# -> 1.44 µs ± 9.72 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

# These are the first 50 sentences from Google's Conceptual Captions dataset (see
# https://ai.google.com/research/ConceptualCaptions/).
TEXTS = [
    "a very typical bus station",
    "sierra looked stunning in this top and this skirt while performing with person at their former university",
    "young confused girl standing in front of a wardrobe",
    "interior design of modern living room with fireplace in a new house",
    "cybernetic scene isolated on white background .",
    "gangsta rap artist attends sports team vs playoff game in the borough .",
    "the jetty : different types of plants to establish a variety of ecosystems .",
    "traditional ornamental floral paisley bandanna .",
    "# of the sports team skates against sports team during their game .",
    "by geographical feature category or in the city - a dome for every environment",
    "a flight was traveling when the animal got free on tuesday night",
    "even though agricultural conditions are not ideal for growing tobacco , there is indigenous production .",
    "us state speaks during a demonstration thursday .",
    "actor arrives for the premiere of the film",
    "celebrities start decorating for the christmas season lifestyle",
    "functions of government : 1 . form a more perfect union",
    "actor attends the premiere of season",
    "american football player on the field during joint training camp .",
    "companies have gone to court for the right to lie",
    "all shots by by person and rider shots can be found on his website .",
    "photo of a deer and wildfire",
    "high angle view of a businessman lying on a table and singing",
    "this is real fast food !",
    "safe deposit with money around it on a white background photo",
    "the giraffe before he was shot dead then autopsied in the presence of the zoo 's visitors , despite an online petition to save him signed by thousands of animal lovers",
    "dunes lay the blueprint for the back nine .",
    "portrait of a smiling woman stroking her dog lying on couch",
    "young business woman on a bench",
    "american football player looks downfield during the second half of a football game against sports team",
    "... and local people to deliver a new bridge",
    "actor arrives to the premiere",
    "funny animals of the week , animal pictures",
    "see the inspiring way this woman documented her travels on her prosthetic leg",
    "the sign promises as much as the glorious blue sky .",
    "architectural details of a bridge",
    "people tour and enjoy the public park during summer",
    "interesting 1930 's poster for a cosmetic company with stores .",
    "racecar driver steers his car during video game subject .",
    "vintage elegant floral card with frame decorated with black and white lilies on a pink background .",
    "heavy snow falls over a snow lined river .",
    "bright living room in the attic",
    "pop artist attends the 3rd annual at guest house",
    "illustration of a map , its flag and a comic balloon with a soccer ball in a not allowed signal",
    "rock artist performs on stage at awards held",
    "green sea turtle isolated on a white background 3d illustration",
    "person , was surprised by the staff",
    "red and white flag on the mast",
    "football player celebrates scoring for football team against football team in the final",
    "concept plug - in hybrid car on display",
    "a pencil drawing of a zebra and her baby .",
]

%timeit tokenizer_fast(TEXTS, return_tensors="np", padding="max_length", max_length=77, return_token_type_ids=False, return_attention_mask=False)
# -> 1.79 ms ± 7.49 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
%timeit tokenizer_instant.tokenize_batch(TEXTS)
# -> 160 µs ± 2.15 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

</details>

**Comment by ArthurZucker** [comment]
Thanks a lot for sharing! Indeed I played around with it and seems a LOT faster 😅 
We do have extra logic with normalizers, pre_tokenizers and added vocabulary which might? explain this but I'll see if I find anything! 
Thanks again for sharing! 🤗
